### PR TITLE
Simulation Interface with Unused Parameter

### PR DESCRIPTION
**Pull request by chrisliebert**
…code

---

This change is [Reviewable](https://reviewable.io/reviews/hgb2/psas-capstone/21)

### CODE REVIEW

File: src/flight_mode/src/i2c.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ pub struct Myi2c {
 
 impl Myi2c {
 
-    pub fn init() -> Result<Myi2c, io::Error> {
+    pub fn init(unused: *const MyPins) -> Result<Myi2c, io::Error> {
```
Comment (BrianBreniser) [inline src/flight_mode/src/i2c.rs:13]: Why is this argument unused? Why include it at all then?

**Comment by chrisliebert** [comment]
Review status: 0 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

_[src/flight_mode/src/i2c.rs, line 13 [r1]](https://reviewable.io:443/reviews/hgb2/psas-capstone/21#-KO2nCqy44qVxUCcHTvC-r1-13:-KO2own9PhhSWu-2ybVS:-758895490) ([raw file](https://github.com/hgb2/psas-capstone/blob/3c7db4341dff5fa2d6c75d94e7c751830fe33ef4/src/flight_mode/src/i2c.rs#L13)):_
<details><summary>

<i>Previously, BrianBreniser (Brian Breniser) wrote…</i></summary>



> Why is this argument unused? Why include it at all then?
> </details>
> In is needed in order for the common main interface to work with test mode.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hgb2/psas-capstone/21)_

**Comment by cwalexand** [comment]
I still disagree with gpio pins being used in i2c regardless of which mode is executing.

**Comment by chrisliebert** [comment]
It is ugly, I will admit however it does not effect the runtime functionality of flight mode. The best alternative I can think of is moving main from common to flight_mode, and writing a different main for test mode. @cwalexand do you have an alternate proposal?

**Comment by cwalexand** [comment]
Yes, I can help. Let's talk about it in the meeting today.

**Pull request closed without merging**